spec/features/staff/users_can_sign_in_spec.rb

```diff
@@ -42,34 +42,140 @@ def log_in_via_form(user, remember_me: false)
+    context "user has a confirmed mobile number" do
+      scenario "successful sign in via header link" do
+        # Given a user with 2FA enabled and a confirmed mobile number exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.now)
+
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
+
+        # Then there should be no link to check my mobile number
+        expect(page).not_to have_link "Check your mobile number is correct"
+
+        otp_at_time_of_login = user.current_otp
+        # And I should receive an OTP to my mobile number
+        expect(notify_client).to have_received(:send_sms).with({
+          phone_number: user.mobile_number,
+          template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
+          personalisation: {otp: otp_at_time_of_login}
+        })
+
+        # When I enter the one-time password before it expires
+        travel 3.minutes do
+          fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+          click_on "Continue"
+        end
+
+        # Then I should be logged in
+        expect(page).to have_link(t("header.link.sign_out"))
+        expect(page).to have_content("Signed in successfully.")
+
+        # And I should be at the home page
+        expect(page).to have_content("You can search by RODA, Delivery Partner, or BEIS identifier, or by the activity's title")
+      end
 
-      # When I log in with that user's email and password
-      visit root_path
-      log_in_via_form(user)
+      scenario "unsuccessful OTP attempt" do
+        # Given a user with 2FA enabled and a confirmed mobile number exists
+        user = create(:administrator, :mfa_enabled, mobile_number_confirmed_at: DateTime.now)
+
+        # When I log in with that user's email and password
+        visit root_path
+        log_in_via_form(user)
 
-      otp_at_time_of_login = user.current_otp
-      # Then I should receive an OTP to my mobile number
-      expect(notify_client).to have_received(:send_sms).with({
-        phone_number: user.mobile_number,
-        template_id: ENV["NOTIFY_OTP_VERIFICATION_TEMPLATE"],
-        personalisation: {otp: otp_at_time_of_login}
-      })
-
-      # When I enter the one-time password before it expires
-      travel 3.minutes do
-        fill_in "Please enter your six-digit verification code", with: otp_at_time_of_login
+        # And I enter an incorrect OTP
+        fill_in "Please enter your six-digit verification code", with: "000000"
         click_on "Continue"
+
+        # Then I should not be logged in
+        expect(page).to have_content("Invalid two-factor verification code")
       end
+    end
 
-      # Then I should be logged in.
-      expect(page).to have_link(t("header.link.sign_out"))
-      expect(page).to have_content("Signed in successfully.")
+    context "user initially entered an incorrect mobile number" do
```

Comment:
I feel this context should be nested with the context below. They're both about if a user doesn't have a mobile number, one is about if they do it successfully, one if they don't?

Which makes me think maybe there should be a test for ensuring you can't edit your mobile number in this flow if you have a confirmed mobile number.

Reply:
Would you object to us adding such a test as an acceptance criterion to https://trello.com/c/2k94LHZX/2391-resetting-a-users-phone-number-for-mfa ?

Reply:
We'll add a test at the model level for the inability to edit confirmed mobile numbers.

Reply:
Hm. It's only possible to edit this were we to craft a POST to `users/session#create` with both a `session[:user_otp_id]` and `user_params[:mobile_number]`. This is difficult to do unless you're actually that user (And CSRF will catch it from the form flow anyway)

Reply:
We have a test that the link to edit doesn't show up when you're confirmed. In light of that we can lock this down at model level on the Resetting MFA card, as it's next to impossible to craft a POST to edit your mobile number through this flow and it's not yet on the admin page.